cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -120,13 +120,21 @@ func newEtcdClient() (client *etcd.Client) {
 			break
 		}
 	}
-	client = etcd.NewClient([]string{*etcd_server})
-	if client == nil {
-		return nil
+	timeout := 10 * time.Second
+	// loop until we have > 0 machines && machines[0] != ""
+	for start := time.Now(); time.Now().Sub(start) < timeout; time.Sleep(time.Second) {
```

Comment:
nit #1: I don't really like this loop.  I think that:

```go
timeout := false
go func() { 
  <-time.After(timeout)
  timeout = true
}()
for !timeout {
...
}
```

or alternately

```go
timeoutChan := time.After(timeout)
tick := time.Tick(time.Second)
for {
  select {
    case <-timeoutChan:
       break;
    case <-tick:
       ...
}
```

Also, use ```time.Since(start)``` instead of ```time.Now().Sub(start)``` it's cleaner.